docs/sources/operations/query-acceleration-blooms.md

```diff
@@ -176,69 +180,45 @@ Example, assuming 4 CPU cores:
-
-## Query sharding
-Query acceleration does not just happen while processing chunks, but also happens from the query planning phase where
-the query frontend applies [query sharding](https://lokidex.com/posts/tsdb/#sharding). 
-Loki 3.0 introduces a new [per-tenant configuration][tenant-limits] flag `tsdb_sharding_strategy` which defaults to computing 
-shards as in previous versions of Loki by using the index stats to come up with the closest power of two that would 
-optimistically divide the data to process in shards of roughly the same size. Unfortunately, 
-the amount of data each stream has is often unbalanced with the rest, 
-therefore, some shards end up processing more data than others.
-
-Query acceleration introduces a new sharding strategy: `bounded`, which uses blooms to reduce the chunks to be 
-processed right away during the planning phase in the query frontend, 
-as well as evenly distributes the amount of chunks each sharded query will need to process.
+For a given stream, the builder will iterate through all the log lines inside its new chunks and build a bloom for the stream.
+In case of changes for a previously processed TSDB file, builders will try to reuse blooms from existing blocks instead of building new ones from scratch.
+The builder converts structured metadata from each log line of each chunk of a stream and appends the hash of each key, value, and key-value pair to the bloom, followed by the hashes combined with the chunk identifier.
+The first set of hashes allows gateways to skip whole streams, while the latter is for skipping individual chunks.
+
+For example, given structured metadata `foo=bar` in the chunk `c6dj8g`, we append to the stream bloom the following hashes: `hash("foo")`, `hash("bar")`, `hash("foo=bar")`, `hash("c6dj8g" + "foo")` ... `hash("c6dj8g" + "foo=bar")`.
+
+### Query sharding
 
+Query acceleration does not just happen while processing chunks, but also happens from the query planning phase where the query frontend applies [query sharding](https://lokidex.com/posts/tsdb/#sharding).
+Loki 3.0 introduces a new [per-tenant configuration][tenant-limits] flag `tsdb_sharding_strategy` which defaults to computing shards as in previous versions of Loki by using the index stats to come up with the closest power of two that would optimistically divide the data to process in shards of roughly the same size.
+Unfortunately, the amount of data each stream has is often unbalanced with the rest, therefore, some shards end up processing more data than others.
+
+Query acceleration introduces a new sharding strategy: `bounded`, which uses blooms to reduce the chunks to be processed right away during the planning phase in the query frontend, as well as evenly distributes the amount of chunks each sharded query will need to process.
+
+[structured metadata]: {{< relref "../get-started/labels/structured-metadata" >}}
```

Comment:
We've moved away from using `relrefs` for linking.  Please see lines 217 through 222 for examples of [how to link to Grafana.com pages](https://grafana.com/docs/writers-toolkit/write/links/#link-to-grafanacom-pages).